smoke-test/tests/cli/graphql_cmd/test_graphql_cli_smoke.py

```diff
@@ -0,0 +1,362 @@
+        exit_code, stdout, stderr = self._run_datahub_cli(
+            ["graphql", "--list-operations"]
+        )
+
+        # Schema discovery might fail due to auth, but command should be recognized
+        assert exit_code in [0, 1], f"Unexpected exit code. stderr: {stderr}"
+        assert "Traceback" not in stderr  # No Python crashes
+        assert (
+            "graphql" not in stderr.lower() or "command not found" not in stderr.lower()
+        )
+
+    def test_graphql_file_path_handling(self):
+        """Test that GraphQL CLI properly handles file path arguments."""
+        with tempfile.NamedTemporaryFile(
+            mode="w", suffix=".graphql", delete=False
+        ) as f:
+            f.write("{ __typename }")
+            temp_path = f.name
+
+        try:
+            # Test with absolute path
+            exit_code, stdout, stderr = self._run_datahub_cli(
+                ["graphql", "--query", temp_path, "--format", "json"]
+            )
+
+            # Should recognize as file path (might fail due to auth/connection)
+            assert exit_code in [0, 1], (
+                f"Unexpected exit code with file path. stderr: {stderr}"
+            )
+            assert "No such file or directory" not in stderr
+            assert "FileNotFoundError" not in stderr
+
+        finally:
+            os.unlink(temp_path)
+
+    def test_graphql_relative_path_handling(self):
+        """Test that GraphQL CLI handles relative paths correctly."""
+        # Create a temporary GraphQL file in a subdirectory
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_path = Path(temp_dir)
+            query_file = temp_path / "test.graphql"
+            query_file.write_text("{ __typename }")
+
+            # Change to parent directory and use relative path
+            original_cwd = os.getcwd()
+            try:
+                os.chdir(temp_path.parent)
+                relative_path = os.path.relpath(str(query_file))
+
+                exit_code, stdout, stderr = self._run_datahub_cli(
+                    ["graphql", "--query", relative_path, "--format", "json"]
+                )
+
+                # Should recognize relative path (might fail due to auth/connection)
+                assert exit_code in [0, 1], (
+                    f"Relative path handling failed. stderr: {stderr}"
+                )
+                assert "No such file or directory" not in stderr
+                assert "FileNotFoundError" not in stderr
+
+            finally:
+                os.chdir(original_cwd)
+
+
+class TestGraphQLCLIIntegration:
+    """Integration tests requiring full DataHub functionality."""
+
+    @pytest.fixture(autouse=True)
+    def setup_datahub(self, auth_session):
+        """Ensure DataHub is running and accessible."""
+        self.auth_session = auth_session
+        wait_for_healthcheck_util(requests)
+
+    def _run_authenticated_graphql(self, args: list[str]) -> tuple[int, str, str]:
```

Comment:
here an example of `run_datahub_cmd` with auth session

https://github.com/datahub-project/datahub/blob/667b7cb12cfdec763bbc9c65e45b2646222a41b0/smoke-test/tests/cli/user_groups_cmd/test_user_cmd.py#L54-L60